galasa-extensions-parent/dev.galasa.auth.couchdb/src/main/java/dev/galasa/auth/couchdb/internal/CouchdbAuthStoreValidator.java

```diff
@@ -6,26 +6,176 @@
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpStatus;
+import org.apache.http.StatusLine;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
 import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.util.EntityUtils;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
 
 import dev.galasa.extensions.common.couchdb.CouchdbBaseValidator;
 import dev.galasa.extensions.common.couchdb.CouchdbException;
 import dev.galasa.extensions.common.api.HttpRequestFactory;
+import dev.galasa.framework.spi.utils.GalasaGson;
 
-public class CouchdbAuthStoreValidator extends CouchdbBaseValidator {
+public class CouchdbAuthStoreValidator extends CouchdbBaseValidator{
 
     private final Log logger = LogFactory.getLog(getClass());
+    private final GalasaGson                         gson               = new GalasaGson();
+    
+    
 
     @Override
     public void checkCouchdbDatabaseIsValid(URI couchdbUri, CloseableHttpClient httpClient, HttpRequestFactory httpRequestFactory) throws CouchdbException {
+
         // Perform the base CouchDB checks
         super.checkCouchdbDatabaseIsValid(couchdbUri, httpClient, httpRequestFactory);
-
         validateDatabasePresent(couchdbUri, CouchdbAuthStore.TOKENS_DATABASE_NAME);
+        checkTokensDesignDocument(httpClient, couchdbUri, 1);
 
         logger.debug("Auth Store CouchDB at " + couchdbUri.toString() + " validated");
+
+    }
+    
+    public void checkTokensDesignDocument( CloseableHttpClient httpClient , URI couchdbUri , int attempts) throws CouchdbException {
+        HttpRequestFactory requestFactory = super.getRequestFactory();
+        HttpGet httpGet = requestFactory.getHttpGetRequest(couchdbUri + "/galasa_tokens/_design/docs");
+
+        String docJson = null;
+        try (CloseableHttpResponse response = httpClient.execute(httpGet)) {
+
+            StatusLine statusLine = response.getStatusLine();
+
+            docJson = EntityUtils.toString(response.getEntity());
+            if (statusLine.getStatusCode() != HttpStatus.SC_OK
+                    && statusLine.getStatusCode() != HttpStatus.SC_NOT_FOUND) {
+                throw new CouchdbException(
+                        "Validation failed of database galasa_tokens design document - " + statusLine.toString());
+            }
+            if (statusLine.getStatusCode() == HttpStatus.SC_NOT_FOUND) {
+                docJson = "{}";
+            }
+        } catch (CouchdbException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new CouchdbException("Validation failed", e);
+        }
+
+        boolean updated = false;
+
+        JsonObject doc = gson.fromJson(docJson, JsonObject.class);
```

Comment:
Can't you unmarshall the json into java beans, rather than go through all this dynamic json manipulation ?

eg: If it's not in the java bean it won't fail, but will ignore fields the java bean doesn't understand.